### PR TITLE
Adding wrapper function that adds instrumentation/metrics when `fetch()` fails

### PR DESCRIPTION
**Pull request by nicholas-ly**
### Description
Originally, if a `fetch()`rejects due to a a badly-formed request URL or a network error, the application metrics aren't updated. This PR addresses that for the `POST /benefit-application` fetch call only and will be applied to other services when approved.

### Checklist
- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/utils/fetch-utils.ts [hunk 1]
```diff
@@ -40,3 +41,24 @@ export function getFetchFn(proxyUrl?: string, timeout?: number) {
...
+ * @param metricPrefix A string prefix used for naming metrics (ex. "http.client.example-api.example-endpoint.gets", "http.client.example-api.example-endpoint.posts").
+ * @param input The URL or request object specifying the target of the fetch request
+ * @param init An object containing additional options for the fetch request (ex. headers, method)
+ * @returns A Promise that resolves to the Response object from the fetch() call
+ * @throws The original error thrown by the underlying fetch() call
+ */
+export async function instrumentedFetch(fetchFn: FetchFunction, metricPrefix: string, input: string | URL, init?: FetchFunctionInit) {
+  try {
+    const response = await fetchFn(input, init);
+    return response;
```
Comment (gregory-j-baker) [inline frontend/app/utils/fetch-utils.ts:58]: I think you should consider looking at the response status here and emitting a metric if it's OK or not.

Comment (nicholas-ly) [reply frontend/app/utils/fetch-utils.ts:60]: Considered and added [here](https://github.com/DTS-STN/canadian-dental-care-plan/compare/feb61527a9b1e8d2be2c21c326302b33115cb5db..dfa6ac08b8ca8eb92cb7d1f713dc6e4d3d7b966f#diff-5f5669eb63caa505348a0900e4f6e7d3ec8f675ea39ebf957d511dccaedcdf4aR59).

File: frontend/app/utils/fetch-utils.ts [hunk 2]
```diff
@@ -40,3 +41,25 @@ export function getFetchFn(proxyUrl?: string, timeout?: number) {
...
+ * Wraps a fetch() function and adds instrumentation/metrics collection
+ *
+ * @param fetchFn An existing fetch() function used to make the actual request
+ * @param metricPrefix A string prefix used for naming metrics (ex. "http.client.example-api.example-endpoint.gets", "http.client.example-api.example-endpoint.posts").
+ * @param input The URL or request object specifying the target of the fetch request
+ * @param init An object containing additional options for the fetch request (ex. headers, method)
+ * @returns A Promise that resolves to the Response object from the fetch() call
+ * @throws The original error thrown by the underlying fetch() call
+ */
+export async function instrumentedFetch(fetchFn: FetchFunction, metricPrefix: string, input: string | URL, init?: FetchFunctionInit) {
```
Comment (sebastien-comeau) [inline frontend/app/utils/fetch-utils.ts:55]: I don't think `fetchFn` is reqired since your function signature already have `fetchFn` arguments.

Comment (nicholas-ly) [reply frontend/app/utils/fetch-utils.ts:55]: `fetchFn` is needed to distinguish whether the consumer passed `global.fetch` or `undiciFetch(..)` for when a proxy is used.

Comment (sebastien-comeau) [reply frontend/app/utils/fetch-utils.ts:55]: Gotcha

**Comment by sebastien-comeau** [comment]
Think renaming `fetch-utils.ts` to `fetch-utils.server.ts` since it only contains server-side functions.

**Comment by nicholas-ly** [comment]
> Think renaming `fetch-utils.ts` to `fetch-utils.server.ts` since it only contains server-side functions.

Good catch - fixed [here](https://github.com/DTS-STN/canadian-dental-care-plan/pull/1780/commits/dd065ce817c2885f5ada545b3be99b88fce48321).

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**